lib/desktop_screens_new/welcome_screen/widgets/sidebar_item.dart

```diff
@@ -0,0 +1,192 @@
+    }
+    if ((item.isEmail == true)) {
+      await _launchInEmail(arguments!['email']);
+      return;
+    }
+
+    if (item.routeName != null && item.routeName != '') {
+      if (item.routeName == DesktopRoutes.DESKTOP_HOME) {
+        await DesktopSetupRoutes.nested_pop();
+        return;
+      }
+      await DesktopSetupRoutes.nested_push(item.routeName,
+          arguments: arguments);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    SizeConfig().init(context);
+    var nestedProvider = Provider.of<NestedRouteProvider>(
+        NavService.navKey.currentContext!,
+        listen: false);
+
+    return Container(
+      child: menuItem.children?.isEmpty ?? true
+          ? InkWell(
+              onTap: () {
+                onTapItem(menuItem);
+              },
+              child: BuildSidebarIconTitle(
+                image: menuItem.image,
+                route: menuItem.routeName ?? "",
+                isSidebarExpanded: isSidebarExpanded,
+                title: menuItem.title,
+                nestedProvider: nestedProvider,
+              ),
+            )
+          : ExpansionTile(
+              childrenPadding: EdgeInsets.zero,
+              tilePadding: EdgeInsets.zero,
+              onExpansionChanged: (value) {
+                onTapItem(menuItem);
+              },
+              maintainState: true,
+              title: BuildSidebarIconTitle(
+                image: menuItem.image,
+                route: menuItem.routeName ?? "",
+                isSidebarExpanded: isSidebarExpanded,
+                title: menuItem.title,
+                nestedProvider: nestedProvider,
+              ),
+              children: menuItem.children!.map((item) {
+                return Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 5),
+                  child: InkWell(
+                    onTap: () {
+                      onTapItem(item);
+                    },
+                    child: BuildSidebarIconTitle(
+                      image: item.image,
+                      route: item.routeName ?? "",
+                      isSidebarExpanded: isSidebarExpanded,
+                      title: item.title,
+                      nestedProvider: nestedProvider,
+                    ),
+                  ),
+                );
+              }).toList(),
+              collapsedBackgroundColor: Colors.transparent,
+              // backgroundColor: Theme.of(context).primaryColor,
+            ),
+    );
+  }
+
+  Future<void> _launchInBrowser(String url) async {
+    try {
+      await launchUrl(
+        Uri(
+          scheme: 'https',
+          path: url,
+        ),
+      );
+    } catch (e) {
+      throw 'Could not launch $url';
+    }
+  }
+
+  Future<void> _launchInEmail(String email) async {
+    await launchUrl(
+      Uri(
+        scheme: 'mailto',
+        path: email,
+      ),
+    );
+  }
+}
+
+class BuildSidebarIconTitle extends StatelessWidget {
```

Comment:
New component should be in a new file.

Reply:
Its a local component that is only used by the main component in that file